src/components/inputs/NumberInput/index.js

```diff
@@ -45,15 +56,40 @@ const NumberInput = ({
     onChange,
     multiline,
     autoComplete,
+    min,
+    max,
 }) => {
+    const { formatMessage } = useSafeIntl();
     const hasErrors = errors.length >= 1;
-    const [formattedValue, setFormattedValue] = useState(formatValue(value));
+    const [formattedValue, setFormattedValue] = useState(
+        formatValue(value, min, max),
+    );
 
     useEffect(() => {
-        const formatted = formatValue(value);
-        setFormattedValue(formatted);
+        const formatted = formatValue(value, min, max, formattedValue);
+        if (
+            formatted !== formattedValue &&
+            formatted < min && // avoiding copy paste of wrong value
+            formatted > max
+        ) {
+            setFormattedValue(formatted);
+        }
     }, [value]);
```

Comment:
you need to add formattedValue to the deps array. The `if` statement will prevent an infinite loop